### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by Keith-Go**
## Ticket
- [feature search](https://trello.com/c/mnaW5Nrw/5-t%C3%ACm-ki%E1%BA%BFm)

## To do
- Search event
- Search user

## Video

https://github.com/luke-nguyen-goldenowl/join_me/assets/152830147/ffd96777-614b-4dc1-b91f-dcbb7ee40652

### CODE REVIEW

File: lib/src/features/search/logic/person_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import 'package:myapp/src/features/search/logic/person_state.dart';
+import 'package:myapp/src/network/domain_manager.dart';
+import 'package:myapp/src/network/model/user/user.dart';
+
+class PersonBloc extends Cubit<PersonState> {
+  PersonBloc({required MUser person}) : super(PersonState(person: person));
+  DomainManager domain = DomainManager();
+  void onPressedFollowHost() async {
+    try {
+      List<String> newFollower = [...state.person.followers!];
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/search/logic/person_bloc.dart]: change null check

File: lib/src/features/search/logic/person_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      final user = GetIt.I<AccountBloc>().state.user.copyWith();
+      List<String> newFollowed = [...user.followed!];
```
Comment (luke-nguyen-goldenowl) [inline lib/src/features/search/logic/person_bloc.dart]: change null check

File: lib/src/network/data/event/event_reference.dart [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+class EventReference extends BaseCollectionReference<MEvent> {
+  EventReference()
+      : super(
+          FirebaseFirestore.instance.collection('events').withConverter<MEvent>(
+                fromFirestore: (snapshot, options) => MEvent.fromMap(
+                    snapshot.data() as Map<String, dynamic>, snapshot.id),
+                toFirestore: (chatRoom, _) => chatRoom.toMap(),
+              ),
+          getObjectId: (e) => e.id!,
```
Comment (luke-nguyen-goldenowl) [inline lib/src/network/data/event/event_reference.dart]: Change null check

File: lib/src/network/data/event/event_reference.dart [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    }
+  }
+
+  Future<MResult<MEvent>> getEvent(String eventId) async {
+    try {
+      UserReference userReference = UserReference();
+      final MResult<MEvent> eventResult = await get(eventId);
+      if (eventResult.isSuccess) {
+        final MResult<MUser> user =
+            await userReference.getUser(eventResult.data!.host!.id);
```
Comment (luke-nguyen-goldenowl) [inline lib/src/network/data/event/event_reference.dart]: change null check for `host`

File: lib/src/network/data/story/story_reference.dart [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  StoryReference()
+      : super(
+          FirebaseFirestore.instance
+              .collection('stories')
+              .withConverter<MStory>(
+                fromFirestore: (snapshot, options) => MStory.fromMap(
+                    snapshot.data() as Map<String, dynamic>, snapshot.id),
+                toFirestore: (chatRoom, _) => chatRoom.toMap(),
+              ),
+          getObjectId: (e) => e.id!,
```
Comment (luke-nguyen-goldenowl) [inline lib/src/network/data/story/story_reference.dart]: Change null check

File: lib/src/network/data/story/story_reference.dart [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+          setObjectId: (e, id) => e.copyWith(id: id),
+        );
+
+  final XFirebaseStorage firebaseStorage = XFirebaseStorage();
+
+  Future<MResult<MStory>> addStory(MStory story) async {
+    try {
+      final image = await firebaseStorage.uploadImage(story.image!, "stories");
```
Comment (luke-nguyen-goldenowl) [inline lib/src/network/data/story/story_reference.dart]: change null check

File: lib/src/network/data/user/user_repository_impl.dart [hunk 7]
```diff
@@ -10,25 +10,52 @@ class UserRepositoryImpl extends UserRepository {
...
-      final result = FirebaseAuth.instance.currentUser;
-      if (result == null) {
+      final result = await usersRef.getUser(id);
+      // final result = FirebaseAuth.instance.currentUser;
+      if (result.isError) {
         return MResult.error('Not user login');
       }
-      final user =
-          MUser(id: result.uid, email: result.email, name: result.displayName);
+      final user = MUser(
```
Comment (luke-nguyen-goldenowl) [inline lib/src/network/data/user/user_repository_impl.dart]: should check isSuccess to be able to check null with "!"

**Comment by Keith-Go** [comment]
resolved

File: lib/src/network/data/user/user_repository_impl.dart [hunk 8]
```diff
@@ -11,19 +11,22 @@ class UserRepositoryImpl extends UserRepository {
...
-        id: result.data!.id,
-        email: result.data!.email,
-        name: result.data!.name,
-        avatar: result.data!.avatar,
-        followers: result.data!.followers,
-        followed: result.data!.followed,
-      );
-      return MResult.success(user);
+
+      if (result.isSuccess) {
```
Comment (luke-nguyen-goldenowl) [inline lib/src/network/data/user/user_repository_impl.dart]: user `else` instead of `isError`

**Comment by luke-nguyen-goldenowl** [comment]
After fixing, you can merge

**Pull request merged**